tests/integration/test_charm.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration test for the content-cache charm."""
+
+import pytest
+from juju.application import Application
+from juju.model import Model
+
+from tests.integration.helpers import CacheTester
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.asyncio
+async def test_charm_start(app: Application, config_app: Application, model: Model) -> None:
+    """
+    arrange: The applications deployed.
+    act: Nothing.
+    assert: The applications in blocked status waiting for integration.
+    """
+    assert len(app.units) == 1
+    unit = app.units[0]
+    assert unit.workload_status_message == "Waiting for integration with config charm"
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.asyncio
+async def test_charm_integrate_with_no_data(
+    app: Application, config_app: Application, model: Model
+) -> None:
+    """
+    arrange: A working application of content-cache charm, with no integrations.
+    act:
+        1. Integrate with a charm that provides configuration.
+        2. Add configuration to the charm that provides data.
```

Comment:
I'm not following what these are? Which one is the integrator charm please?

Reply:
I changed the terminology to `configuration charm` in all the docstrings and code comments of the file.